### PR TITLE
[SPARK-42706][SQL][DOCS] Document the Spark SQL error classes in user-facing documentation.

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to document Spark SQL error classes to [Spark SQL Guide](https://spark.apache.org/docs/latest/sql-programming-guide.html).

- Error Conditions
Screen Shot 2023-03-08 at 8 54 43 PM

- SQLSTATE Codes
Screen Shot 2023-03-08 at 8 54 54 PM

- Error Classes that includes sub-error classes (`INVALID_FORMAT` as an example)
Screen Shot 2023-03-08 at 9 10 22 PM

### Why are the changes needed?

To improve the usability for error messages for Spark SQL.

### Does this PR introduce _any_ user-facing change?

No API change, but yes, it's user-facing documentation.

### How was this patch tested?

Manually built docs and check the contents one-by-one compare to [error-classes.json](https://github.com/apache/spark/blob/master/core/src/main/resources/error/error-classes.json).

### CODE REVIEW

**Comment by itholic** [comment]
cc @srielau @MaxGekk @cloud-fan This is user-facing documentation for SQL error classes.

**Comment by MaxGekk** [comment]
@itholic Did you write those files manually, or generated by a script?

File: docs/sql-error-conditions-sqlstates.md [hunk 1]
```diff
@@ -0,0 +1,720 @@
...
+
+While many `SQLSTATE` values are prescribed by the SQL standard, others are common in the industry, specific to Spark.
+
+For an ordered list of error classes see: [Error Conditions in Spark SQL](sql-error-conditions.html)
+
+Spark SQL uses the following `SQLSTATE` classes:
+
+## Class `0A`: feature not supported
+
+<table class="table">
```
Comment (srielau) [inline docs/sql-error-conditions-sqlstates.md:36]: Isn't .md supposed to take care of the HTML conversion of the table?

**Review by srielau — APPROVED** [review]

File: docs/sql-error-conditions-sqlstates.md (hunk 1 above)
Comment (itholic) [reply docs/sql-error-conditions-sqlstates.md:36]: I am not entirely sure, but it seems like Jekyll(the tool used for building documentation in OSS Spark) has specific rules for conversion. I checked the table may not be built properly if we don't use the html tag here.

**Comment by itholic** [comment]
> @itholic Did you write those files manually, or generated by a script?

I did both. I used the script, and for the parts that didn't print correctly, I fixed them manually by checking the page one by one. Since the script is not very polished for now, let me submit it as a follow-up task if we aim for automation in the future.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @itholic and @srielau for review.

**Comment by gatorsmile** [comment]
@MaxGekk  should we merge it to 3.4?

**Comment by MaxGekk** [comment]
@gatorsmile I stoped merging PRs related to error classes after RC0. Need to re-check carefully which error classes are in branch-3.4 already. @itholic Please, backport the changes to branch-3.4.

**Comment by cloud-fan** [comment]
It's probably ok to mention non-existing errors in the doc in 3.4

**Comment by itholic** [comment]
3.4: https://github.com/apache/spark/pull/40433

**Pull request closed without merging**